### PR TITLE
Highlight Basic/AttCA ambiguity in definitions and verification procedures

### LINKED ISSUE #1042: FIDO U2F supports Attestation CA (AttCA)?
[Section 8.6](https://w3c.github.io/webauthn/#fido-u2f-attestation) describes FIDO U2F attestation statement format. The supported attestation types includes _Basic_ and _AttCA_.
But, there is no description regarding _AttCA_ in case of U2F although TPM format has such description for _AttCA_. Also, there is no case which returns _AttCA_ after verifying U2F attestation.

I know that there were discussions related to this from Google, I'm not sure whether decision is made to support _AttCA_ or not.

So, We have following options.

1. Remove _AttCA_ from the U2F supported attestation types
2. Add descriptions of _AttCA_ cases to [Section 8.6](https://w3c.github.io/webauthn/#fido-u2f-attestation). Also, modify _AttCA_ description in [Section 6.4.3](https://w3c.github.io/webauthn/#sctn-attestation-types) by adding U2F cases

### PR DESCRIPTION
**Pull request by emlun**
This fixes #1042.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1054.html" title="Last updated on Sep 18, 2018, 4:47 PM GMT (721c507)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1054/5bdcae4...721c507.html" title="Last updated on Sep 18, 2018, 4:47 PM GMT (721c507)">Diff</a>

### CODE REVIEW

**Comment by nadalin** [comment]
@selfissued Please review

File: index.bs [hunk 1]
```diff
@@ -3213,6 +3216,10 @@ WebAuthn supports multiple attestation types:
     Note: This concept typically leads to multiple attestation certificates. The attestation certificate requested most recently
         is called "active".
 
+    Note: [=Attestation statements=] of type [=AttCA=] use the same data structure as [=attestation statements=] of type
+    [=Basic=], so the two attestation types are in general not distinguishable without externally provided knowledge about the
```
Comment (selfissued) [inline index.bs]: put commas around "in general"

Comment (emlun) [reply index.bs]: Done!

File: index.bs [hunk 2]
```diff
@@ -4006,7 +4016,10 @@ This attestation statement format is used with FIDO U2F authenticators using the
     1. Let |verificationData| be the concatenation of (0x00 || |rpIdHash| ||
         |clientDataHash| || |credentialId| || |publicKeyU2F|) (see [=Section 4.3=] of [[!FIDO-U2F-Message-Formats]]).
     1. Verify the |sig| using |verificationData| and |certificate public key| per [[!SEC1]].
-    1. If successful, return attestation type [=Basic=] with the [=attestation trust path=] set to |x5c|.
+    1. Optionally, inspect |x5c| and consult externally provided knowledge to determine whether |attStmt| conveys a [=Basic=] or
+        [=AttCA=] attestation.
+    1. If successful, return attestation type [=Basic=] or [=AttCA=], or an implementation-specific value expressing uncertainty,
```
Comment (equalsJeffH) [inline index.bs]: suggest:
return attestation type =>  return an implementation-specific value representing  [=attestation type=]

File: index.bs [hunk 3]
```diff
@@ -4006,7 +4016,10 @@ This attestation statement format is used with FIDO U2F authenticators using the
...
+        and [=attestation trust path=] |x5c|.
```
Comment (equalsJeffH) [inline index.bs]: suggest:
and [=attestation trust path=] |x5c| => and in either case also return [=attestation trust path=] |x5c|

File: index.bs [hunk 4]
```diff
@@ -3191,6 +3191,9 @@ The initial list of specified [=attestation statement formats=] is in [[#defined
 
 WebAuthn supports multiple attestation types:
```
[the remark this answers is not in the document]
Comment (emlun) [reply index.bs]: Sounds good, thanks!

File: index.bs [hunk 5]
```diff
@@ -3617,7 +3624,10 @@ implementable by [=authenticators=] with limited resources (e.g., secure element
             - Verify that |attestnCert| meets the requirements in [[#packed-attestation-cert-requirements]].
             - If |attestnCert| contains an extension with OID 1.3.6.1.4.1.45724.1.1.4 (`id-fido-gen-ce-aaguid`) verify that the
                 value of this extension matches the <code>[=aaguid=]</code> in |authenticatorData|.
-            - If successful, return attestation type [=Basic=] and [=attestation trust path=] |x5c|.
+            - Optionally, inspect |x5c| and consult externally provided knowledge to determine whether |attStmt| conveys a
+                [=Basic=] or [=AttCA=] attestation.
+            - If successful, return attestation type [=Basic=] or [=AttCA=], or an implementation-specific value expressing
```
[the remark this answers is not in the document]
Comment (emlun) [reply index.bs]: How about instead making this change in [§6.4.2. Attestation Statement Formats][attform]?

>[...]
>The [verification] procedure returns either:
>- An error indicating that the attestation is invalid, or
>- **An implementation-specific value representing an** attestation type, and the trust path. [...]

[attform]: https://w3c.github.io/webauthn/#attestation-formats

File: index.bs [hunk 6]
```diff
@@ -3617,7 +3624,10 @@ implementable by [=authenticators=] with limited resources (e.g., secure element
...
+                uncertainty, and [=attestation trust path=] |x5c|.
```
[the remark this answers is not in the document]
Comment (emlun) [reply index.bs]: Done!

File: index.bs (hunk 3 above)
Comment (emlun) [reply index.bs]: Done!

File: index.bs [hunk 7]
```diff
@@ -3191,6 +3191,9 @@ The initial list of specified [=attestation statement formats=] is in [[#defined
 
 WebAuthn supports multiple attestation types:
 
+Note: [=Attestation types=] are not expressed in any data structure defined in this specification. Any values representing
+[=attestation types=] are internal and implementation-specific to the [=[RP]=].
```
[the remark this answers is not in the document]
Comment (emlun) [reply index.bs]: Thanks! I did the first part of this; see https://github.com/w3c/webauthn/pull/1054#discussion_r218033959 for my response to the second part.

File: index.bs (hunk 5 above)
[the remark this answers is not in the document]
Comment (emlun) [reply index.bs]: See PR #1070

File: index.bs (hunk 2 above)
Comment (emlun) [reply index.bs]: See above.

File: index.bs [hunk 8]
```diff
@@ -3213,6 +3216,10 @@ WebAuthn supports multiple attestation types:
     Note: This concept typically leads to multiple attestation certificates. The attestation certificate requested most recently
         is called "active".
 
+    Note: [=Attestation statements=] of type [=AttCA=] use the same data structure as [=attestation statements=] of type
+    [=Basic=], so the two attestation types are, in general, not distinguishable without externally provided knowledge about the
+    [=attestation certificates=] conveyed in the [=attestation statement=].
```
[the remark this answers is not in the document]
Comment (equalsJeffH) [reply index.bs]: "cumbersome" -- excellent description ;-)

File: index.bs (hunk 7 above)
[the remark this answers is not in the document]
Comment (equalsJeffH) [reply index.bs]: ok, thx ;)

**Comment by nadalin** [comment]
@emlun This should be ready to merge now

File: index.bs (hunk 5 above)
[the remark this answers is not in the document]
Comment (emlun) [reply index.bs]: All right, will do.

[the remark this answers is not in the document]
Comment (emlun) [reply index.bs]: My take on your requested changes are in PR #1071.

**Comment by emlun** [comment]
Some additional changes requested by @equalsJeffH are implemented in meta-PR #1071.

**Comment by equalsJeffH** [comment]
thx, meta-PR #1071 approved :)

**Comment by emlun** [comment]
I think we're ready to merge unless @selfissued wants to review the [additional changes since his review][diff].

[diff]: https://github.com/w3c/webauthn/pull/1054/files/845e92022e51c2c451bfeb80fc4bb69b73e85fcc..721c50710ab2454c0cf9349325593fc4e19ea07d

File: index.bs (hunk 8 above)
[the remark this answers is not in the document]
Comment (emlun) [reply index.bs]: :smile:

**Comment by nadalin** [comment]
@emlun Please merge

**Pull request merged**